common/service/dynamicconfig/constants.go

```diff
@@ -51,6 +51,7 @@ var keys = map[Key]string{
 	EnableGlobalDomain:       "system.enableGlobalDomain",
 	EnableNewKafkaClient:     "system.enableNewKafkaClient",
 	EnableVisibilitySampling: "system.enableVisibilitySampling",
+	EnableKafkaCommon:        "system.enableKafkaCommon",
```

Comment:
I would recommend to make it more specific for each feature: enableKafkaVisibility / enableKafkaInsight. Then we will have better control over each feature. Also EnableKafkaCommon is too generic as feature flag